sympy/solvers/ode.py

```diff
@@ -5930,7 +5897,24 @@ def _remove_coefficient(expr, x):
         # then they will need to be multiplied by sufficient x to make them so.
         # This function DOES NOT do that (it doesn't even look at the
         # homogeneous equation).
-        retdict['trialset'] = _get_trial_set(expr, x)
+        temp_set = set([])
+        cartesian_prod = set([])
+        if expr.is_Add:
+            for i in expr.args:
+                act = _get_trial_set(i,x)
+                if eq_homogeneous is not S.Zero:
+                    f = func.func
+                    while any([expand(*[sub_func_doit(eq_homogeneous, f(x), ts)]).is_zero for ts in act]):
```

Comment:
We don't need `expand(*[...])` as it can just be `expand(...)`

Reply:
@oscarbenjamin the function returns generator so thats why i used list here